mmv1/products/kms/api.yaml

```diff
@@ -162,6 +162,168 @@ objects:
         'Creating a key':
           'https://cloud.google.com/kms/docs/creating-keys#create_a_key'
       api: 'https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys'
+  - !ruby/object:Api::Resource
+    name: 'CryptoKeyVersion'
+    base_url: '{{crypto_key}}/cryptoKeyVersions'
+    create_url: '{{crypto_key}}/cryptoKeyVersions'
+    self_link: '{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A `CryptoKeyVersion` represents an individual cryptographic key, and the associated key material.
+    parameters: 
+      - !ruby/object:Api::Type::String
+        name: 'cryptoKey'
+        required: true
+        input: true
+        url_param_only: true
+        description: |
+          The name of the cryptoKey associated with the CryptoKeyVersions.
+          Format: `'projects/{{project}}/locations/{{location}}/keyRings/{{keyring}}/cryptoKeys/{{cryptoKey}}'`
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The resource name for this CryptoKeyVersion.
+        output: true
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
+        description: |
+          The current state of the CryptoKeyVersion.
+        values:
+          - "PENDING_GENERATION"
+          - "ENABLED"
+          - "DISABLED"
+          - "DESTROYED"
+          - "DESTROY_SCHEDULED"
+          - "PENDING_IMPORT"
+          - "IMPORT_FAILED"
+      - !ruby/object:Api::Type::Enum
+        name: 'protectionLevel'
+        output: true
+        description: |
+          The ProtectionLevel describing how crypto operations are performed with this CryptoKeyVersion.
+        values:
+          - "PROTECTION_LEVEL_UNSPECIFIED"
+          - "SOFTWARE"
+          - "HSM"
+          - "EXTERNAL"
+          - "EXTERNAL_VPC"
+      - !ruby/object:Api::Type::Time
+        name: 'createTime'
+        description: |
+          The time at which this cryptoKeyVersion was created
+        output: true
+      - !ruby/object:Api::Type::Time
+        name: 'generateTime'
+        description: |
+          The time this CryptoKeyVersion key material was generated
+        output: true
+      - !ruby/object:Api::Type::Time
+        name: 'destroyEventTime'
```

Comment:
The `destroy*Time` fields are probably not useful here since they would only be accessible / have values on a resource that is being deleted, which would already not be in Terraform (so the values wouldn't be accessible).

Reply:
Agreed, I will remove them.